crates/core/src/test_utils/deploy_helpers.rs

```diff
@@ -98,12 +98,29 @@ pub fn get_contract(filename: &str) -> CompactContractBytecode {
     let dot_sol = format!("{filename}.sol");
     let dot_json = format!("{filename}.json");
 
-    let foundry_default_out = load_config().out;
-    let compiled_solidity_path = std::path::Path::new(&foundry_default_out).join(dot_sol).join(dot_json);
-    let compiled_solidity_path_from_root = root_project_path!(&compiled_solidity_path);
+    let foundry_output_path: PathBuf = {
+        let foundry_output_path = std::env::var("FOUNDRY_OUTPUT_PATH").expect(
+            "Expected a FOUNDRY_OUTPUT_PATH environment variable, set up your .env file or use \
+             `./scripts/make_with_env.sh test`",
+        );
+
+        // We search firstly for existence for absolute path
+        let foundry_output_path_absolute = PathBuf::from(&foundry_output_path);
+        if foundry_output_path_absolute.try_exists().unwrap() {
```

Comment:
in what case can we panic here?

Reply:
File permissions, etc.

Overall, try_exists will resolve with an Ok value of true or false in normal scenarios, but it was not able to do its investigation, because of some other reason { i.e couldn't read because of permissions, etc then it will panic. }